daemon/api_notices.go

```diff
@@ -188,15 +188,17 @@ func sanitizeNoticeTypesFilter(queryTypes []string, r *http.Request) ([]state.No
 		}
 		// No types were specified, populate with notice types snap can view
 		// with its connected interface.
-		ucred, iface, err := ucrednetGetWithInterface(r.RemoteAddr)
+		ucred, ifaces, err := ucrednetGetWithInterface(r.RemoteAddr)
 		if err != nil {
 			return nil, err
 		}
 		if ucred.Socket == dirs.SnapdSocket {
 			// Not connecting through snapd-snap.socket, should have read-access to all types.
 			return nil, nil
 		}
-		types = allowedNoticeTypesForInterface(iface)
+		for _, iface := range ifaces {
+			types = append(types, allowedNoticeTypesForInterface(iface)...)
```

Comment:
What happens if a snap has two interfaces with overlapping allowed types, would that mean duplicate types? maybe we need a test for this

Reply:
It would mean duplicate types, good catch. I've added an `alreadySeen` map to make sure each type is only added once.